test-data/unit/check-literal.test

```diff
@@ -3149,3 +3149,18 @@ err_code = -TWO
 if bool():
     err_code = -THREE
 [builtins fixtures/float.pyi]
+
+[case testAliasForEnumTypeAsLiteral]
+from typing_extensions import Literal
+
+class Out:
+    class In:
+        class Inner:
+            pass
+
+O = Out
+I = Out.In
+OI = O.In
+x: O.In
+y: I.Inner
+z: OI.Inner
```

Comment:
Please add a `reveal_type()` for all three variables. (Also please keep newline at the end of file.)